src/helm/benchmark/runner.py

```diff
@@ -185,4 +216,84 @@ def run_one(self, run_spec: RunSpec):
+            metric_names_from_schema.add("predicted_index")
+
+        stats_by_trial: Dict[Tuple[str, Optional[PerturbationDescription], int], Dict[str, float]] = defaultdict(dict)
+        for original_stats in per_instance_stats:
+            stats_dict: Dict[str, float] = {
+                original_stat.name.name: cast(float, original_stat.mean)
+                for original_stat in original_stats.stats
+                if original_stat.name.name in metric_names_from_schema
+            }
+            key = (original_stats.instance_id, original_stats.perturbation, original_stats.train_trial_index)
+            stats_by_trial[key].update(stats_dict)
+
+        predictions = []
+        for request_state in scenario_state.request_states:
+            assert request_state.instance.id is not None
+            assert request_state.result is not None
+
+            predicted_text = (
+                request_state.result.completions[0].text
+                if request_state.result is not None or request_state.result.completions
+                else None
+            )
+            mapped_output = (
+                request_state.output_mapping.get(predicted_text.strip()) if request_state.output_mapping else None
+            )
+
+            stats_key = (
+                request_state.instance.id,
+                request_state.instance.perturbation,
+                request_state.train_trial_index,
+            )
+            predictions.append(
+                SlimPrediction(
+                    instance_id=request_state.instance.id,
+                    perturbation=request_state.instance.perturbation,
+                    train_trial_index=request_state.train_trial_index,
+                    predicted_text=predicted_text,
+                    predicted_text_prefix_to_hide=self._compute_predicted_text_prefix_to_hide(
+                        request_state, run_spec.adapter_spec
+                    ),
+                    mapped_output=mapped_output,
+                    reference_index=request_state.reference_index,
+                    stats=stats_by_trial[stats_key],
+                )
+            )
+        write(
+            os.path.join(run_path, "predictions.json"),
+            json.dumps(list(map(asdict_without_nones, predictions)), indent=2),
+        )
+
+    def _compute_predicted_text_prefix_to_hide(
+        self, request_state: RequestState, adapter_spec: AdapterSpec
+    ) -> Optional[str]:
+        method = adapter_spec.method
+        if method.startswith("multiple_choice_separate_"):
+            return request_state.instance.input
+        elif method == "language_modeling":
+            if request_state.result is not None and request_state.result.completions:
+                tokens = request_state.result.completions[0].tokens
+                if tokens:
+                    first_token = tokens[0]
+                    if not first_token.top_logprobs:
```

Comment:
When does this happen?

Reply:
I'm not sure to be honest. The [`Token` docs](https://github.com/stanford-crfm/helm/blob/main/src/helm/common/request.py#L76) do not explain this. This simply replicates the logic [here](https://github.com/stanford-crfm/helm/blob/main/src/helm/benchmark/static/benchmarking.js#L605).

Reply:
Should we drop this if condition?

Reply:
Added warning.